prow/slack/reporter/reporter.go

```diff
@@ -77,7 +82,11 @@ func (sr *slackReporter) GetName() string {
 }
 
 func (sr *slackReporter) ShouldReport(pj *v1.ProwJob) bool {
-	config := sr.config()
+	config, err := sr.config(pj.Spec.Refs)
+	if err != nil {
+		sr.logger.WithField("prowjob", pj.Name).Error(err)
```

Comment:
I don't actually think using an error here makes sense, as its a perfectly valid configuration to not have a default. What about e.G. returning an empty but non-nil Slackconfig? That should result in no reporting being done

Reply:
> What about e.G. returning an empty but non-nil Slackconfig?

I switched it so that instead of returning a `(**SlackReporter, error)` in `config(...)` it is simply a `SlackReporter`.